Experiment with runs-on

gsmet opened the pull request:
See https://runs-on.com for more information.

gsmet committed:
Experiment with runs-on

See https://runs-on.com for more information.

gastaldi commented on .github/matrix-jvm-tests.json:
```diff
@@ -3,20 +3,23 @@
   "maven_opts": "-Xmx2g -XX:MaxMetaspaceSize=1g",
-  "os-name": "ubuntu-latest"
+  "os-name": "ubuntu-latest",
+  "runs-on": "family=m7a.large/image=ubuntu24-full-x64"
```
Don't you need to specify the `${{ github.run_id }}`?

Eg.

```yaml
runs-on:
 - runs-on=${{ github.run_id }}
 - family= m7a.large
 - image= ubuntu24-full-x64
```

gsmet replied:
It's done in the workflow.

gastaldi replied:
Ah doh, this is the json, thanks

gsmet committed:
Update version of MongoDB used for testing to 7.0

It is not the latest (8.0 was released in November) but it's probably
the most common used and also it's consistent with the version used as a
container.

Hopefully, this will be compatible with Ubuntu 22 and 24.

gsmet committed:
Drop max-parallel as it's not supported by runs-on for now

gsmet committed:
Switch to m7i.xlarge instances to get more memory

gsmet committed:
Disable Develocity caching to get real timings

gsmet committed:
Test caching

gsmet commented:
This can be closed. A much improved approach has been merged a while ago.
